### PR TITLE
Internal: Allow hosting to replace more promotions [ED-13518]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**

### CODE REVIEW

File: modules/promotions/admin-menu-items/form-submissions-promotion-item.php [hunk 1]
```diff
@@ -21,15 +19,15 @@ public function get_page_title() {
 		return esc_html__( 'Submissions', 'elementor' );
 	}
 
-	public function get_promotion_title() {
+	protected function get_promotion_title(): string {
```
Comment (davseve) [inline modules/promotions/admin-menu-items/form-submissions-promotion-item.php:22]: I see that the access modifiers were changed only in this file.
Maybe it is better to change all the access modifiers for the desired function in the rest of the desired classes?

File: modules/promotions/admin-menu-items/custom-icons-promotion-item.php [hunk 2]
```diff
@@ -21,14 +20,14 @@ public function get_page_title() {
...
-		echo sprintf(
+		return sprintf(
 			/* translators: %s: br  */
 			esc_html( 'Enjoy creative freedom %s with Custom Icons', 'elementor' ),
 			'<br />'
 		);
 	}
 
-	public function set_list() {
+	public function get_content_lines(): array {
```
Comment (davseve) [inline modules/promotions/admin-menu-items/custom-icons-promotion-item.php:30]: If we are already changing the API, why not to change the access modifiers as well?

File: modules/promotions/admin-menu-items/custom-icons-promotion-item.php [hunk 3]
```diff
@@ -38,11 +37,11 @@ public function set_list() {
 		];
 	}
 
-	public function get_cta_url() {
+	public function get_cta_url(): string {
 		return 'https://go.elementor.com/go-pro-custom-icons/';
 	}
 
-	public function get_video_url() {
+	public function get_video_url(): string {
```
Comment (davseve) [inline modules/promotions/admin-menu-items/custom-icons-promotion-item.php:44]: Please Change the access modifier

**Review by davseve — DISMISSED** [review]

**Review by davseve — APPROVED** [review]

**Comment by louiswol94** [comment]
@TzviRabinovitch @davseve @Barcov 
I don't know the full story, but so far I think it is unsafe that we changed these function names in Core while this base class is being used in Pro (core/pro dependency).

**Comment by Barcov** [comment]
> @TzviRabinovitch @davseve @Barcov I don't know the full story, but so far I think it is unsafe that we changed these function names in Core while this base class is being used in Pro (core/pro dependency).

Yes agreed it's what I've been calling for prior to our decision to break the core/pro dependancy on Friday. But even then changing signatures is not ideal. David and I discussed keeping get_description() and injecting the <li> in. I implemented this on Friday's merge for Pro, and David would have implemented this in core, but the discussed changed never made it into core. Some of these PRO changes seems to have vanished by Monday morning ? So now we have a core dependancy again.

**Pull request merged**